backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/command/InterimInvoiceFlatrateTermCreateCommand.java

```diff
@@ -0,0 +1,253 @@
+			interimInvoiceFlatrateTermBL.updateQuantities(interimInvoiceFlatrateTerm);
+		}
+	}
+
+	private InterimInvoiceFlatrateTerm getOrCreateInterimInvoiceOverview(final I_C_Flatrate_Term flatrateTerm)
+	{
+		final InterimInvoiceFlatrateTerm interimInvoiceFlatrateTerm = interimInvoiceFlatrateTermDAO.getInterimInvoiceForFlatrateTermAndOrderLineId(FlatrateTermId.ofRepoId(flatrateTerm.getC_Flatrate_Term_ID()), orderLineId);
+		if (interimInvoiceFlatrateTerm != null && interimInvoiceFlatrateTermDAO.isInterimInvoiceStillUsable(interimInvoiceFlatrateTerm))
+		{
+			return interimInvoiceFlatrateTerm;
+		}
+		else
+		{
+			return createInterimInvoiceOverview(flatrateTerm);
+		}
+	}
+
+	@NonNull
+	private I_C_Flatrate_Term getOrCreateFlatrateTerm()
+	{
+		Check.assumeEquals(conditions.getType_Conditions(), X_C_Flatrate_Conditions.TYPE_CONDITIONS_InterimInvoice);
+		final I_C_BPartner bpartner = bpartnerDAO.getById(bpartnerId);
+		final Optional<I_C_Flatrate_Term> existingFlatrateTerm = getExistingFlatrateTerm(bpartner);
+		return existingFlatrateTerm.orElseGet(() -> createFlatrateTerm(bpartner));
+	}
+
+	private Optional<I_C_Flatrate_Term> getExistingFlatrateTerm(final @NonNull I_C_BPartner bpartner)
+	{
+		return flatrateDAO.retrieveTerms(bpartner, conditions)
+				.stream()
+				.filter(term -> dateFrom.equals(term.getStartDate()))
+				.filter(term -> dateTo.equals(term.getEndDate()))
+				.findFirst();
+	}
+
+	@NonNull
+	private I_C_Flatrate_Term createFlatrateTerm(final I_C_BPartner bpartner)
+	{
+		return FlatrateTermCreator.builder()
+				.bPartners(Collections.singleton(bpartner))
+				.orgId(orderLine.getOrgId())
+				.conditions(conditions)
+				.ctx(ctx)
+				.product(product)
+				.startDate(dateFrom)
+				.endDate(dateTo)
+				.isCompleteDocument(true)
+				.build()
+				.createTermsForBPartners()
+				.stream()
+				.findFirst()
+				.orElseThrow(() -> new AdempiereException("No flatrate term created"));
+	}
+
+	private InterimInvoiceFlatrateTerm createInterimInvoiceOverview(@NonNull final I_C_Flatrate_Term flatrateTerm)
+	{
+		final I_C_Order order = orderDAO.getById(orderLine.getOrderId());
+
+		final UomId productUomId = UomId.ofRepoId(product.getC_UOM_ID());
+		final Quantity zeroQty = Quantitys.createZero(productUomId);
+		final CurrencyId currencyId = CurrencyId.ofRepoId(order.getC_Currency_ID());
+		final InterimInvoiceFlatrateTerm interimInvoiceFlatrateTerm = InterimInvoiceFlatrateTerm.builder()
+				.flatrateTermId(FlatrateTermId.ofRepoId(flatrateTerm.getC_Flatrate_Term_ID()))
+				.calendarId(interimInvoiceSettings.getHarvestingCalendarId())
+				.orderLineId(orderLineId)
+				.qtyOrdered(uomConversionBL.convertQuantityTo(orderLine.getOrderedQty(), productId, productUomId))
+				.qtyDelivered(zeroQty)
+				.qtyInvoiced(zeroQty)
+				.productId(productId)
+				.uomId(productUomId)
+				.priceActual(Money.of(BigDecimal.ZERO, currencyId))
```

Comment:
leave it like this now,
but just want to make u aware that we have `Money.zero(currencyId)`, for convenience.
Use which one u feel it convenient

Reply:
done.